kafka_utils/kafka_manual_throttle/main.py

```diff
@@ -0,0 +1,191 @@
+
+    True if the options are valid, False otherwise.
+
+    :opts : the command line options
+    :returns : bool
+    """
+    if opts.clear:
+        if opts.leader_throttle is not None:
+            print("Error: --clear cannot be used with --leader-throttle")
+            return False
+        if opts.follower_throttle is not None:
+            print("Error: --clear cannot be used with --follower-throttle")
+            return False
+    else:
+        if opts.leader_throttle < 0:
+            print("Error: --leader-throttle must be >= 0")
+            return False
+        if opts.follower_throttle < 0:
+            print("Error: --follower-throttle must be >= 0")
+            return False
+    return True
+
+
+def apply_throttles(zk, brokers, leader_throttle, follower_throttle):
+    """
+    Apply new leader/follower replication throttles to the given brokers
+
+    :zk : a ZK client
+    :brokers : a collection of broker ids
+    :leader_throttle : new leader replication throttle (in B/s)
+    :follower_throttle : new follower replication throttle (in B/s)
+    """
+    for broker_id in brokers:
+        write_throttle(zk, broker_id, str(leader_throttle), str(follower_throttle))
+
+
+def clear_throttles(zk, brokers):
+    """
+    Clear replication throttles for the given brokers.
+
+    :zk : a ZK client
+    :brokers : a collection of broker ids
+    """
+    for broker_id in brokers:
+        write_throttle(zk, broker_id, None, None)
+
+
+def write_throttle(zk, broker_id, leader_throttle, follower_throttle):
+    """
+    Write leader/follower replication throttle rates for a given broker.
+
+    If a given replication throttle is None, the configuration will be cleared
+    from the broker instead of applying a new one.
+
+    More details can be found in:
+        https://kafka.apache.org/documentation/#rep-throttle
+
+    :zk : a ZK client
+    :broker_id : broker to change replication throttles for
+    :leader_throttle : new leader replication throttle (in B/s) or None
+    :follower_throttle : new follower replication throttle (in B/s) or None
+    """
+    config = zk.get_broker_config(broker_id)
+
+    if leader_throttle is not None:
+        config[LEADER_THROTTLE_RATE_CONFIGURATION] = leader_throttle
+    else:
+        config.pop(LEADER_THROTTLE_RATE_CONFIGURATION, None)
+
+    if follower_throttle is not None:
+        config[FOLLOWER_THROTTLE_RATE_CONFIGURATION] = follower_throttle
+    else:
+        config.pop(FOLLOWER_THROTTLE_RATE_CONFIGURATION, None)
+
+    zk.set_broker_config(broker_id, config)
+
+
+def run():
+    opts = parse_opts()
+    if opts.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.WARN)
+
+    if not validate_opts(opts):
+        sys.exit(1)
+
+    cluster_config = config.get_cluster_config(
+        opts.cluster_type,
+        opts.cluster_name,
+        opts.discovery_base_path,
+    )
+
+    print("Applying throttles")
```

Comment:
Let's log the previous throttles and the new throttles. Both in the big byte numbers but also in human-readable format (you can use https://github.com/xolox/python-humanfriendly to get a friendly byte size)

Reply:
Good idea! I've updated the PR, the command now prints throttle values before and after.
I've also added a `--read-only` option to allow us to just have a look at throttles without actually changing them